solution/build.sh

```diff
@@ -0,0 +1,136 @@
+#!/bin/bash
+PWD=$(pwd)
+BUILD_ROOT=${PWD}/_build
+ISO_ROOT=${BUILD_ROOT}/root
+IMAGES_ROOT=${ISO_ROOT}/images
+
+PRODUCT_NAME=Zenko
+PRODUCT_LOWERNAME=zenko
+BUILD_TIMESTAMP=$(date -u +"%Y-%m-%dT%H:%M:%SZ")
+BUILD_HOST=$(hostname)
+
+VERSION_SHORT=$(git describe --abbrev=0)
+VERSION_FULL=${VERSION_SHORT}-dev
+GIT_REVISION=$(git describe --long --always --tags --dirty)
+ISO=${BUILD_ROOT}/${PRODUCT_LOWERNAME}-${VERSION_FULL}.iso
+
+DOCKER=docker
+DOCKER_OPTS=
+DOCKER_SOCKET=unix:///var/run/docker.sock
+HARDLINK=hardlink
+OPERATOR_SDK=operator-sdk
+OPERATOR_SDK_OPTS=
+SKOPEO=skopeo
+SKOPEO_OPTS="--override-os linux --insecure-policy"
+
+# grab our dependencies from our deps.txt file as an array
+readarray -t DEP_IMAGES < deps.txt
+# add the CR dependencies by joining image: and tag: lines
+readarray -t CR_IMAGES < <(grep image: zenko-version-cr.yaml | awk '{print $2}')
+readarray -t CR_TAGS < <(grep tag: zenko-version-cr.yaml | awk '{print $2}')
+
+for index in ${!CR_IMAGES[@]}; do
+	DEP_IMAGES+=(${CR_IMAGES[index]}:${CR_TAGS[index]})
+done
+
+function clean()
+{
+	rm -rf ${BUILD_ROOT}
+}
+
+function mkdirs()
+{
+	mkdir -p ${ISO_ROOT}
+}
+
+function copy_yamls()
+{
+	OPERATOR_TAG=$(grep /zenko-operator: deps.txt | awk -F ':' '{print $2}')
+	sed "s/ZENKO_OPERATOR_TAG/${OPERATOR_TAG}/" config.yaml > ${ISO_ROOT}/config.yaml
+	cp -R -f operator/ ${ISO_ROOT}/operator
+}
+
+function gen_product_txt()
+{
+	cat > ${ISO_ROOT}/product.txt <<EOF
+NAME=${PRODUCT_NAME}
+VERSION=${VERSION_FULL}
+SHORT_VERSION=${VERSION_SHORT}
+GIT=${GIT_REVISION}
+DEVELOPMENT_RELEASE=1
+BUILD_TIMESTAMP=${BUILD_TIMESTAMP}
+BUILD_HOST=${BUILD_HOST}
+EOF
+}
+
+function gen_operator_yaml()
+{
+	OPERATOR_IMAGE=$(grep /zenko-operator: deps.txt)
+	# we need to escape / with \/ in our sed command
+	sed "s/REPLACE_IMAGE/${OPERATOR_IMAGE//\//\\/}/" operator.yaml > ${ISO_ROOT}/operator/operator.yaml	
+}
+
+function copy_image()
+{
+	IMAGE_NAME=${1##*/}
+	FULL_PATH=${IMAGES_ROOT}/${IMAGE_NAME/:/\/}
+	mkdir -p ${FULL_PATH}
+	${SKOPEO} ${SKOPEO_OPTS} copy \
+		--format v2s2 --dest-compress \
+		--src-daemon-host ${DOCKER_SOCKET} \
+		docker-daemon:${1} \
+		dir:${FULL_PATH}
+}
+
+function dedupe()
+{
+	${HARDLINK} -c ${IMAGES_ROOT}
+}
+
+function build_registry_config()
+{
+	docker run \
+    	 	--name static-oci-registry \
+	 	--mount type=bind,source=${ISO_ROOT}/images,destination=/var/lib/images \
+		--mount type=bind,source=${ISO_ROOT},destination=/var/run \
+	 	--rm \
+    		docker.io/nicolast/static-container-registry:latest \
```

Comment:
Is `nicolast` the current reference repository for the registry? Might be good to keep a TODO around to think about changing it if it is temporary, and maybe use a shell variable for it.

Reply:
https://github.com/NicolasT/static-container-registry doesn't look temporary, but maybe @NicolasT has a recommendation?

Reply:
Indeed it's not a temporary repository, and is being included as a submodule of scality/metalk8s as well (since it's a rather basic Python script, maybe you don't need the Docker container to run it?).

Reply:
The docker container avoids having python as a dependency, and is really just as convenient for this case. ( just in case it adds a dependency and versions for python, i can avoid entirely. )